services/wallet/cryptocompare.go

```diff
@@ -0,0 +1,128 @@
+package wallet
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+)
+
+type Coin struct {
+        Id                      string                 `json:"Id"`
+        Name                    string                 `json:"Name"`
+        Symbol                  string                 `json:"Symbol"`
+        Description             string                 `json:"Description"`
+        TotalCoinsMined         float64                `json:"TotalCoinsMined"`
+        AssetLaunchDate         string                 `json:"AssetLaunchDate"`
+        AssetWhitepaperUrl      string                 `json:"AssetWhitepaperUrl"`
+        AssetWebsiteUrl         string                 `json:"AssetWebsiteUrl"`
+        BuiltOn                 string                 `json:"BuiltOn"`
+        SmartContractAddress    string                 `json:"SmartContractAddress"`
+}
+
+type MarketCoinValues struct {
+        MKTCAP                      string                 `json:"MKTCAP"`
+        HIGHDAY                     string                 `json:"HIGHDAY"`
+        LOWDAY                      string                 `json:"LOWDAY"`
+        CHANGEPCTHOUR               string                 `json:"CHANGEPCTHOUR"`
+        CHANGEPCTDAY                string                 `json:"CHANGEPCTDAY"`
+        CHANGEPCT24HOUR             string                 `json:"CHANGEPCT24HOUR"`
+}
+
+type CoinsContainer struct {
+        Data map[string]Coin `json:"Data"`
+}
+
+type MarketValuesContainer struct {
+        Display map[string]map[string]MarketCoinValues `json:"Display"`
+}
+
+func fetchCryptoComparePrices(symbols []string, currency string) (map[string]float64, error) {
+	httpClient := http.Client{Timeout: time.Minute}
+
+	url := fmt.Sprintf("https://min-api.cryptocompare.com/data/pricemulti?fsyms=%s&tsyms=%s&extraParams=Status.im", strings.Join(symbols, ","), currency)
```

Comment:
When we were using cryptocompare's API on status-teller-network, we noticed that there was a limit in the number of symbols you could request before the API started returning errors. We solved it by querying the API in batches of 20 currencies:

https://github.com/status-im/status-teller-network/blob/b72ed063404da276db24a180ad47d2f4bb917438/src/js/features/prices/saga.js#L64-L68

It's probably worth testing if this is still the case by querying a lot of tokens

Reply:
hey @richard-ramos I verified and we are querying over 49 symbols and there is so no error printed when making the api call :(
Maybe they updated their limit?